src/abstract_penalty.jl

```diff
@@ -1,12 +1,12 @@
 # Code based on code from OnlineStats by Josh day (see LICENSE.md)
 
-abstract Penalty
+abstract ParamCost
```

Comment:
There is no clear consensus for the terms. For example in the theory I based the implementation on Loss function is a type of cost function (which I don't particularly like). In other words, it's up to us to decide.